File: 2_1 ai assistant persona/n8-assistant-v2/scripts/export-carousel.mjs
import { chromium } from 'playwright';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, '..');
const HTML_FILE = path.join(ROOT, 'public/carousels/hormozi-enhance-offer-carousel.html');
const OUT_DIR = path.join(ROOT, 'public/carousels/export/hormozi-enhance');
const TOTAL_SLIDES = 8;

const WIDTH = 1080;
const HEIGHT = 1350;

async function exportSlides() {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: WIDTH, height: HEIGHT },
    deviceScaleFactor: 2,
  });
  const page = await context.newPage();

  const fileUrl = 'file://' + HTML_FILE;
  console.log('Opening:', fileUrl);

  await page.goto(fileUrl, { waitUntil: 'networkidle' });

  // Ждём загрузку шрифтов и рендер
  await page.waitForTimeout(1500);

  for (let i = 1; i <= TOTAL_SLIDES; i++) {
    const hash = `#slide-${i}`;
    await page.goto(fileUrl + hash, { waitUntil: 'networkidle' });
    await page.waitForTimeout(800);

    const outPath = path.join(OUT_DIR, `slide-${String(i).padStart(2, '0')}.png`);
    await page.screenshot({ path: outPath, width: WIDTH, height: HEIGHT, fullPage: false });
    console.log(`✓ Saved ${outPath}`);
  }

  await browser.close();
  console.log('\nDone! All 8 slides exported to', OUT_DIR);
}

exportSlides().catch((err) => {
  console.error('Export failed:', err);
  process.exit(1);
});

File: 2_1 ai assistant persona/n8-assistant-v2/scripts/export-carousel.py
#!/usr/bin/env python3
"""Export carousel slides to PNG using Playwright (Python)."""
import asyncio
import os
import sys
from pathlib import Path

from playwright.async_api import async_playwright

ROOT = Path(__file__).parent.parent.resolve()
HTML_FILE = ROOT / "public/carousels/hormozi-enhance-offer-carousel.html"
OUT_DIR = ROOT / "public/carousels/export/hormozi-enhance"
TOTAL_SLIDES = 8
WIDTH = 1080
HEIGHT = 1350


async def export_slides():
    OUT_DIR.mkdir(parents=True, exist_ok=True)

    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(
            viewport={"width": WIDTH, "height": HEIGHT},
            device_scale_factor=2,
        )
        page = await context.new_page()

        file_url = "file://" + str(HTML_FILE)
        print(f"Opening: {file_url}")

        await page.goto(file_url, wait_until="networkidle")
        # Ждём загрузку шрифтов и рендер
        await page.wait_for_timeout(15000)

        for i in range(1, TOTAL_SLIDES + 1):
            hash_fragment = f"#slide-{i}"
            await page.goto(file_url + hash_fragment, wait_until="networkidle")
            await page.wait_for_timeout(800)

            out_path = OUT_DIR / f"slide-{i:02d}.png"
            await page.screenshot(
                path=str(out_path),
                clip={"x": 0, "y": 0, "width": WIDTH, "height": HEIGHT},
            )
            print(f"✓ Saved {out_path}")

        await browser.close()
        print(f"\nDone! All {TOTAL_SLIDES} slides exported to {OUT_DIR}")


if __name__ == "__main__":
    asyncio.run(export_slides())